File: dotnet/cafdemalihapa/Directories/Hosting/Feature/FrontEnd/FrontEndSecondary/FrontEndSecondaryDirectory.cs
using cafdemalihapa.Files;
using cafdemalihapa.Names;

namespace cafdemalihapa.Directories.Hosting.Feature.FrontEnd.FrontEndSecondary
{
    public class FrontEndSecondaryDirectory(
            SecondaryApplication secondaryApplication,
            FrontEndDirectory frontEndDirectory
        )
    {
        public string GetName()
        {
            var frontEndDirectoryName = frontEndDirectory.GetName();
            var secondaryApplicationName = secondaryApplication.GetName();
            var name = $"{frontEndDirectoryName}.{secondaryApplicationName}";
            return name;
        }

        public string GetPath()
        {
            var frontEndDirectoryPath = frontEndDirectory.GetPath();
            var name = GetName();

            var frontEndSecondaryDirectoryPath = Path.Combine(frontEndDirectoryPath, name);
            return frontEndSecondaryDirectoryPath;
        }

        public void Create()
        {
            var path = GetPath();
            if(!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
            }
        }

        public void CreateFiles()
        {
            var path = GetPath();
            FileService.CreateNumberedFiles(path);
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/FrontEnd/FrontEndSecondary/FrontEndSecondaryDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.FrontEnd.FrontEndSecondary

open System.IO
open cafdemwimu.console.Files
open cafdemwimu.console.Names
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd

type FrontEndSecondaryDirectory(secondaryApplication: SecondaryApplication, frontEndDirectory: FrontEndDirectory) =
    member _.GetName() =
        let frontEndDirectoryName = frontEndDirectory.GetName()
        let secondaryApplicationName = secondaryApplication.GetName()
        let name = $"{frontEndDirectoryName}.{secondaryApplicationName}"
        name

    member this.GetPath() =
        let frontEndDirectoryPath = frontEndDirectory.GetPath()
        let name = this.GetName()
        let frontEndSecondaryDirectoryPath = Path.Combine(frontEndDirectoryPath, name)
        frontEndSecondaryDirectoryPath

    member this.Create() =
        let path = this.GetPath()
        if not (Directory.Exists(path)) then
            Directory.CreateDirectory(path) |> ignore

    member this.CreateFiles() =
        let path = this.GetPath()
        FileService.CreateNumberedFiles path
